tools/topsites.py

```diff
@@ -135,47 +142,90 @@ def parse_site(site):
             site_row.ranking = rank
 
 
-def build_uri(country_code, start_ranking):
+def build_request(country_code, start_ranking):
     """Build Alexa top site URI with given country code and start ranking."""
-    # Build query string
-    query_string = build_query_string(country_code, start_ranking)
+    # Prepare timestamp & datestamp
+    nowtime = datetime.datetime.utcnow()
+    timestamp = nowtime.strftime(ATS_DATEFORMAT_AWS)
+    datestamp = nowtime.strftime(ATS_DATEFORMAT_CREDENTIAL)
+
+    # Build request
+    canonical_query = build_query_string(country_code, start_ranking)
+    canonical_headers = 'host:{host}\nx-amz-date:{amzdate}\n'.format(
+        host=ATS_SERVICE_ENDPOINT,
+        amzdate=timestamp)
+    payload_hash = get_sha256_hex("")
+
+    canonical_request = 'GET\n{service_uri}\n{query}\n{headers}\n{signed_headers}\n{payload_hash}'.format(
+        service_uri=ATS_SERVICE_URI,
+        query=canonical_query,
+        headers=canonical_headers,
+        signed_headers=ATS_SIGNED_HEADERS,
+        payload_hash=payload_hash)
```

Comment:
It would be probably easier to read and understand by getting read of `\n`.

```python
canonical_request = """
GET
{service_uri}
{query}
{headers}
{signed_headers}
{payload_hash}""".format(
        service_uri=ATS_SERVICE_URI,
        query=canonical_query,
        headers=canonical_headers,
        signed_headers=ATS_SIGNED_HEADERS,
        payload_hash=payload_hash)
```

but maybe it makes the signing more difficult. 
There are other places in the code with `\n`. 

@MDTsai I can do it when merging.